go/interpreter/lfvm/hash_cache_test.go

```diff
@@ -0,0 +1,210 @@
+				0x49, 0x77, 0x5c, 0xd7, 0x18, 0x75, 0xc4, 0xd0, 0xab, 0x17, 0x08,
+				0xa3, 0x80, 0xe0, 0x3f, 0xef, 0xc3, 0xa2, 0x8a, 0xa2, 0x48, 0x31}},
+	}
+
+	code := Code{Instruction{STOP, 0x0000}}
+
+	for _, test := range tests {
+		t.Run(fmt.Sprintf("hash(%v)", test.data), func(t *testing.T) {
+			cache := newHashCache(1<<16, 1<<18)
+			ctxt := &context{
+				code:  code,
+				stack: NewStack(),
+			}
+			hash := cache.hash(ctxt, test.data)
+			if hash != test.want {
+				t.Fatalf("hash(%v) = %x, want %x", test.data, hash, test.want)
+			}
+		})
+
+	}
+}
+
+func TestHashCache_getHash32CachedEntry(t *testing.T) {
+	cache := newHashCache(1<<16, 1<<18)
+	ctxt := &context{
+		code:  Code{Instruction{STOP, 0x0000}},
+		stack: NewStack(),
+	}
+	data1 := []byte{0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0a,
+		0x0b, 0x0c, 0x0d, 0x0e, 0x0f, 0x10, 0x11, 0x12, 0x13, 0x14, 0x15,
+		0x16, 0x17, 0x18, 0x19, 0x1a, 0x1b, 0x1c, 0x1d, 0x1e, 0x1f}
+	data2 := bytes.Clone(data1)
+	data2[31] = 0x01
+	data3 := bytes.Clone(data1)
+	data3[31] = 0x02
+	hash1 := cache.hash(ctxt, data1)
+	hash2 := cache.hash(ctxt, data2)
+	hash3 := cache.hash(ctxt, data3)
+	entry1 := cache.hash(ctxt, data1)
+	if entry1 != hash1 {
+		t.Fatalf("second cal for same data should return same hash. want %v but got %v", hash1, entry1)
+	}
+	entry3 := cache.hash(ctxt, data3)
+	if entry3 != hash3 {
+		t.Fatalf("second cal for same data should return same hash. want %v but got %v", hash3, entry3)
+	}
+	entry2 := cache.hash(ctxt, data2)
+	if entry2 != hash2 {
+		t.Fatalf("second cal for same data should return same hash. want %v but got %v", hash2, entry2)
+	}
+}
+
+func TestHashCache_getHash64CachedEntry(t *testing.T) {
+	cache := newHashCache(1<<16, 1<<18)
+	ctxt := &context{
+		code:  Code{Instruction{STOP, 0x0000}},
+		stack: NewStack(),
+	}
+	data1 := []byte{0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0a,
+		0x0b, 0x0c, 0x0d, 0x0e, 0x0f, 0x10, 0x11, 0x12, 0x13, 0x14, 0x15,
+		0x16, 0x17, 0x18, 0x19, 0x1a, 0x1b, 0x1c, 0x1d, 0x1e, 0x1f, 0x20,
+		0x21, 0x22, 0x23, 0x24, 0x25, 0x26, 0x27, 0x28, 0x29, 0x2a, 0x2b,
+		0x2c, 0x2d, 0x2e, 0x2f, 0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36,
+		0x37, 0x38, 0x39, 0x3a, 0x3b, 0x3c, 0x3d, 0x3e, 0x3f,
+	}
+	data2 := bytes.Clone(data1)
+	data2[63] = 0x01
+	data3 := bytes.Clone(data1)
+	hash1 := cache.hash(ctxt, data1)
+	hash2 := cache.hash(ctxt, data2)
+	hash3 := cache.hash(ctxt, data3)
+	entry1 := cache.hash(ctxt, data1)
+	if entry1 != hash1 {
+		t.Fatalf("second cal for same data should return same hash. want %v but got %v", hash1, entry1)
+	}
+	entry3 := cache.hash(ctxt, data3)
+	if entry3 != hash3 {
+		t.Fatalf("second cal for same data should return same hash. want %v but got %v", hash3, entry3)
+	}
+	entry2 := cache.hash(ctxt, data2)
+	if entry2 != hash2 {
+		t.Fatalf("second cal for same data should return same hash. want %v but got %v", hash2, entry2)
+	}
+}
+
+func TestHashCache_useCacheTail32(t *testing.T) {
```

Comment:
I think those tests are unnecessarily convoluted and seem to be mainly focused on pushing coverage up. These tests are not sufficient to test the actual requirements on the hash cache.